package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,587 @@
+                    break
+            else:
+                raise NoDataError("Provide at least a position, velocity"
+                                  " or force group in the h5md file.")
+        except ValueError:
+            raise IOError from None
+
+        self._frame = frame
+        ts = self.ts
+        particle_group = self._particle_group
+        ts.frame = frame
+
+        # fills data dictionary from 'observables' group
+        # Note: dt is not read into data as it is not decided whether
+        # Timestep should have a dt attribute (see Issue #2825)
+        self._copy_to_data()
+
+
+        # Sets frame box dimensions
+        # Note: H5MD files must contain 'box' group in each particle group
+        if 'edges' in particle_group['box']:
+            ts._unitcell[:] = particle_group['box/edges/value'][frame, :]
+        else:
+            # sets ts.dimensions = [0, 0, 0, 0, 0, 0]
+            ts._unitcell[:] = None
+
+        # set the timestep positions, velocities, and forces with
+        # current frame dataset
+        if self._has['position']:
+            ts.positions = self._get_frame_dataset('position')
+        if self._has['velocity']:
+            ts.velocities = self._get_frame_dataset('velocity')
+        if self._has['force']:
+            ts.forces = self._get_frame_dataset('force')
+
+        if self.convert_units:
+            self._convert_units()
+
+        return ts
+
+    def _read_next_timestep(self):
+        """read next frame in trajectory"""
+        return self._read_frame(self._frame + 1)
+
+    def _get_frame_dataset(self, dataset):
+        """retrieves dataset array at current frame"""
+
+        frame_dataset = self._particle_group[
+                        dataset]['value'][self._frame, :]
+        n_atoms_now = frame_dataset.shape[0]
+        if n_atoms_now != self.n_atoms:
+            raise ValueError("Frame {} has {} atoms but the initial frame"
+                             " has {} atoms. MDAnalysis is unable to deal"
+                             " with variable topology!"
+                             "".format(self._frame,
+                                       n_atoms_now,
+                                       self.n_atoms))
+        return frame_dataset
+
+    def _convert_units(self):
+        """converts time, position, velocity, and force values if they
+        are not given in MDAnalysis standard units
+
+        See https://userguide.mdanalysis.org/1.0.0/units.html
+        """
+
+        self.ts.time = self.convert_time_from_native(self.ts.time)
+
+        if self._has['position']:
+            self.convert_pos_from_native(self.ts.positions)
+            self.convert_pos_from_native(self.ts.dimensions[:3])
+
+        if self._has['velocity']:
+            self.convert_velocities_from_native(self.ts.velocities)
+
+        if self._has['force']:
+            self.convert_forces_from_native(self.ts.forces)
+
+
+    def _set_translated_units(self):
```

Comment:
The set/check units method could come after `__init__`, closer to where they are used.